### ISSUE TITLE
[Question] How to perform batch inference with this model?

### ISSUE THREAD

**Issue by thangquang09** [kehanlu/DeSTA2.5-Audio#14]
First of all, thank you for sharing this great work.

I am currently trying to run inference on a large dataset. The current README / demo code only shows how to process a single audio file at a time. Could you please provide a code snippet or guidance on how to perform batch inference correctly?

Specifically, I am interested in how to handle the data collator (padding/attention masks) for audio inputs in a batch to maximize throughput.

Thanks in advance!

**Comment by kehanlu** [comment]
Hello @thangquang09 Thank you for interesting in our work!

1. The demo code is for easy hugging face usage. The `DeSTA.generate()` is actually just a wrapper for standard `ll_model.generate()`. I believe with proper batch preparation (passing input_embeds instead of input_ids), it is similar to standard batch inference. (We haven't implemented it because of limited bandwidth)

2. The closest example is from our trainer, we do use this for batch inference with Pytorch lightning. You can use the trainer or borrow some code snippet to the `DeSTA.generate()`.

https://github.com/kehanlu/DeSTA2.5-Audio/blob/5503dfcb8c07ddc608ef705542f0f9b94cc99de8/desta/trainer/pl_trainer.py#L108-L136

https://github.com/kehanlu/DeSTA2.5-Audio/blob/5503dfcb8c07ddc608ef705542f0f9b94cc99de8/desta/models/modeling_desta25.py#L379-L410

3. Another tip is you can cache the ASR transcriptions, then passing them in the message:
```python
messages = [
    {
        "role": "system",
        "content": "Focus on the audio clips and instructions."
    },
    {
        "role": "user",
        "content": "<|AUDIO|>\nDescribe this audio.",
        "audios": [{
            "audio": "/path/to/audio.wav",  # Path to your audio file
            "text": "Your cached transcription"
        }]
    }
]
```

**Comment by thangquang09** [comment]
Thank you for the helpful response!

On a related note, regarding production serving: Have you or the team explored serving DeSTA with high-throughput frameworks like vLLM or something else? 

Thanks again!

**Comment by kehanlu** [comment]
Hello @thangquang09, thank you for your suggestion.

Since we are currently focusing on new research direction, production serving is not our priority (though I believe our model are capable for production deployment). I would love to have further discussion. 😄

**Comment by kehanlu** [comment]
Hi @thangquang09,
We're currently working on the vLLM implementation over on the [vllm](https://github.com/kehanlu/DeSTA2.5-Audio/tree/vllm) branch! 
Please see: [docs/vllm_usage.md](https://github.com/kehanlu/DeSTA2.5-Audio/blob/vllm/docs/vllm_usage.md)
We’d love for you to give it a try and share any feedback!
